ministryofjustice/laa-review-criminal-legal-aid
# CRIMAPP-87 data analyst can download return reasons

## Description of change

- allow data analysts only to download the source data for the return reasons report
- make table headers sticky
- standardise authorisation failure response for authenticated users
- data analysts are shown monthly reports by default
- reporting dashboard links to the latest complete report (rather than the current report)
- paginate temporal reports
- show total number of returns in reporting period

## Link to relevant ticket
[CRIMAPP-87](https://dsdmoj.atlassian.net/browse/CRIMAPP-87)

## Notes for reviewer
See discussion in ticket
requires: https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/194
## Screenshots of changes (if applicable)

### Before changes:
<img width="788" alt="Screenshot 2023-12-28 at 11 21 54" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/34935/2e30f9c2-9030-4aca-96ee-5d81fc34d4aa">

### After changes:
<img width="1026" alt="Screenshot 2023-12-28 at 11 18 12" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/34935/f96f0093-9a6b-4c5a-abbb-93a6c3c90070">

<img width="1134" alt="Screenshot 2023-12-28 at 11 17 36" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/34935/57e4de5d-bc7d-4140-a8e1-d722c5659fb4">

<img width="1288" alt="Screenshot 2023-12-28 at 11 16 00" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/34935/655527ee-c23b-4464-aa8a-2813c8b2766e">

## How to manually test the feature
Log in as a supervisor, confirm you cannot download the return reasons report
Log in as a data analyst, confirm you can download the return reasons report


[CRIMAPP-87]: https://dsdmoj.atlassian.net/browse/CRIMAPP-87?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
